sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -272,3 +321,44 @@ impl TransportWrapper {
         Ok(())
     }
 }
+
+/// Given an pin/uart/spi/i2c port name, if the name is a known
+/// alias, return the underlying name/number, otherwise return the
+/// string as is.
+fn map_name(map: &HashMap<String, String>, name: &str) -> String {
+    let name = name.to_uppercase();
+    match map.get(&name) {
+        Some(v) => {
+            if v.eq(&name) {
+                name
+            } else {
+                map_name(map, v)
+            }
+        }
+        None => name,
+    }
+}
+
+/// Certain transports may want to declare that they do not support a particular pin and that it
+/// should be ignored, even though its name is mentioned in e.g. generic strapping configurations.
+/// (Absent any such declaration, it would result in an error to mention strappings of a pin that
+/// is not in fact supported.)
+struct NullPin;
+
+impl GpioPin for NullPin {
+    fn read(&self) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn write(&self, _value: bool) -> Result<()> {
+        Ok(())
+    }
+
+    fn set_mode(&self, _mode: PinMode) -> Result<()> {
+        Ok(())
+    }
+
+    fn set_pull_mode(&self, _mode: PullMode) -> Result<()> {
+        Ok(())
+    }
+}
```

Comment:
Should we consider using a NullPin to be at least a warnable problem? (ie: add `log::warn!` statements to these functions)?

Reply:
That is a good suggestion, done.